src/handlers/clean/xform.test.ts

```diff
@@ -0,0 +1,333 @@
+      const i = { d: "d" };
+      const o = { d: "d", a: ["d"] };
+      const xformConfig = {
+        a: ["!Path", "$.d"],
+      };
+      expect(xform(xformConfig)(i)).toEqual(o);
+    });
+
+    test("copy value to nested key", () => {
+      const i = { d: "d" };
+      const o = { d: "d", a: { b: ["d"] } };
+      const xformConfig = {
+        "a.b": ["!Path", "$.d"],
+      };
+      expect(xform(xformConfig)(i)).toEqual(o);
+    });
+
+    test("copy nested value to nested key", () => {
+      const i = { d: { e: "e" } };
+      const o = { d: { e: "e" }, a: { b: ["e"] } };
+      const xformConfig = {
+        "a.b": ["!Path", "$..e"],
+      };
+      expect(xform(xformConfig)(i)).toEqual(o);
+    });
+
+    test("copy nested, complex value to nested key", () => {
+      const i = { d: { e: [1, 2, 3] } };
+      const o = { d: { e: [1, 2, 3] }, a: { b: [[1, 2, 3]] } };
+      const xformConfig = {
+        "a.b": ["!Path", "$..e"],
+      };
+      expect(xform(xformConfig)(i)).toEqual(o);
+    });
+
+    test("returns an empty array for empty paths", () => {
+      const res = xform({ a: ["!Path", "$.b"], c: ["!Path", "$.d"] })({
+        d: undefined,
+      });
+      expect(res).toEqual({ a: [], c: [] });
+    });
+  });
+
+  describe("!Equals", () => {
+    test("compares primitives", () => {
+      expect(
+        xform({
+          a: ["!Equals", 1, 1],
+        })({})
+      ).toEqual({ a: true });
+
+      expect(
+        xform({
+          a: ["!Equals", 1, 2],
+        })({})
+      ).toEqual({ a: false });
+
+      expect(
+        xform({
+          a: ["!Equals", "thing", "thing"],
+        })({})
+      ).toEqual({ a: true });
+
+      expect(
+        xform({
+          a: ["!Equals", "thingA", "thingB"],
+        })({})
+      ).toEqual({ a: false });
+
+      expect(
+        xform({
+          a: ["!Equals", true, true],
+        })({})
+      ).toEqual({ a: true });
+
+      expect(
+        xform({
+          a: ["!Equals", true, false],
+        })({})
+      ).toEqual({ a: false });
+    });
+
+    describe("compares arrays", () => {
+      test("matching arrays in the same order", () => {
+        expect(
+          xform({
+            a: ["!Equals", [1, 2, 3], [1, 2, 3]],
+          })({})
+        ).toEqual({ a: true });
+      });
+
+      test("matching arrays in different orders", () => {
+        expect(
+          xform({
+            b: ["!Equals", [1, 3, 2], [1, 2, 3]],
+          })({})
+        ).toEqual({ b: false });
+      });
+
+      test("non-matching arrays ", () => {
+        expect(
+          xform({
+            c: ["!Equals", [1, 2, 3], [1, 2, 4]],
+          })({})
+        ).toEqual({ c: false });
+      });
+
+      test("matching arrays in different orders with ignoreOrder option on", () => {
+        expect(
+          xform({
+            d: ["!Equals", [1, 3, 2], [1, 2, 3], { ignoreOrder: true }],
+          })({})
+        ).toEqual({ d: true });
+      });
```

Comment:
Maybe also compare arrays of different lengths, e.g. 
[1, 2, 3], [1, 2, 3, 4]
and
[1, 2, 3, 4], [1, 2, 3]

Reply:
Thanks, added!